cypress/e2e/editing/editing.spec.ts

```diff
@@ -55,14 +90,213 @@ describe("Editing", () => {
+        });
+    });
+
+    it("should render 'View Source' button in developer mode on the message edit history dialog", () => {
+        cy.visit("/#/room/" + roomId);
+
+        // Send "Message"
+        sendEvent(roomId);
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            // Edit "Message" to "Massage"
+            editLastMessage("Massage");
+
+            // Assert that the edit label is visible
+            cy.get(".mx_EventTile_edited").should("be.visible");
+
+            clickEditedMessage("Massage");
+        });
+
+        cy.get(".mx_Dialog").within(() => {
+            // Assert that the original message is rendered
+            cy.get(".mx_MessageEditHistoryDialog li:nth-child(3)").within(() => {
+                // Assert that "View Source" is not rendered
+                cy.get(".mx_EventTile .mx_EventTile_line")
+                    .realHover()
+                    .contains(".mx_AccessibleButton", "View Source")
+                    .should("not.exist");
+            });
+
+            // Close the dialog
+            cy.get("[aria-label='Close dialog']").click();
+        });
+
+        // Enable developer mode
+        cy.setSettingValue("developerMode", null, SettingLevel.ACCOUNT, true);
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            clickEditedMessage("Massage");
+        });
+
+        cy.get(".mx_Dialog").within(() => {
+            // Assert that the edited message is rendered
+            cy.get(".mx_MessageEditHistoryDialog li:nth-child(2)").within(() => {
+                // Assert that "Remove" button for the original message is rendered
+                cy.get(".mx_EventTile .mx_EventTile_line")
+                    .realHover()
+                    .contains(".mx_AccessibleButton", "Remove")
+                    .should("exist");
+
+                clickButtonViewSource();
+            });
+
+            // Assert that view source dialog is rendered
+            cy.get(".mx_ViewSource").within(() => {
+                // Close the dialog
+                cy.get("[aria-label='Close dialog']").click();
+            });
+
+            // Assert that the original message is rendered
+            cy.get(".mx_MessageEditHistoryDialog li:nth-child(3)").within(() => {
+                // Assert that "Remove" button for the original message does not exist
+                cy.get(".mx_EventTile .mx_EventTile_line")
+                    .realHover()
+                    .contains(".mx_AccessibleButton", "Remove")
+                    .should("not.exist");
+
+                clickButtonViewSource();
+            });
+
+            // Assert that view source dialog is rendered
+            cy.get(".mx_ViewSource").within(() => {
+                // Close the dialog
+                cy.get("[aria-label='Close dialog']").click();
+            });
         });
 
+        // Disable developer mode
+        cy.setSettingValue("developerMode", null, SettingLevel.ACCOUNT, false);
```

Comment:
This should not be necessary as we start with a fresh account every time.

Reply:
That is right. It was removed by https://github.com/matrix-org/matrix-react-sdk/pull/10406/commits/952f3706e4685e2211ae6d1d5f80bd179e3442d9.